### PR TITLE
GeoSvc: Add constantAsString, EnableGeant4Geo, getDetector, deprecated lcdd

### PR DESCRIPTION
**Pull request by andresailer**
BEGINRELEASENOTES

* GeoSvc: add constantAsString implementation
* GeoSvc: add "EnableGeant4Geo" to disable the creation of the Geant4Geometry if it is not needed
* GeoSvc: add getDetector function, marking lcdd() as deprecated
* Replaced use of lcdd() in all places
* SimG4Interface: Removed IGeoSvc.h file, use k4Interface instead

ENDRELEASENOTES

This is part of key4hep/k4MarlinWrapper#27

### CODE REVIEW

File: Detector/DetComponents/src/GeoSvc.cpp [hunk 1]
```diff
@@ -82,13 +84,17 @@ StatusCode GeoSvc::buildDD4HepGeo() {
   return StatusCode::SUCCESS;
 }
 
-dd4hep::Detector* GeoSvc::lcdd() { return (m_dd4hepgeo); }
+dd4hep::Detector* GeoSvc::lcdd() { return m_dd4hepgeo; }
+
+dd4hep::Detector* GeoSvc::getDetector() { return m_dd4hepgeo; }
 
-dd4hep::DetElement GeoSvc::getDD4HepGeo() { return (lcdd()->world()); }
+dd4hep::DetElement GeoSvc::getDD4HepGeo() { return (m_dd4hepgeo->world()); }
```
Comment (andresailer) [inline Detector/DetComponents/src/GeoSvc.cpp:91]:
```suggestion
dd4hep::DetElement GeoSvc::getDD4HepGeo() { return m_dd4hepgeo->world(); }
```

File: Detector/DetComponents/src/GeoSvc.h [hunk 2]
```diff
@@ -50,6 +53,9 @@ class GeoSvc : public extends<Service, IGeoSvc> {
   Gaudi::Property<std::vector<std::string>> m_xmlFileNames{this, "detectors", {}, "Detector descriptions XML-files"};
   /// mapping of sensitive detector names
   Gaudi::Property<std::map<std::string, std::string>> m_sensitive_types{this, "sensitiveTypes", {{"tracker", "SimpleTrackerSD"}, {"calorimeter", "SimpleCalorimeterSD"}}};
+  /// Wether to create the geant4 geometry or not
```
Comment (andresailer) [inline Detector/DetComponents/src/GeoSvc.h:56]:
```suggestion
  /// Whether to create the geant4 geometry or not
```

**Comment by andresailer** [comment]
@kjvbrt Please have a look at this

**Comment by kjvbrt** [comment]
Hi,

I'm getting crashes at `m_runManager.Initialize();`, when running tests locally at Alma9/nightly stack.

```
#6  0x00007fc6c0421281 in G4RunManagerKernel::DefineWorldVolume(G4VPhysicalVolume*, bool) () from /cvmfs/sw
-nightlies.hsf.org/key4hep/releases/2023-09-03/x86_64-almalinux9-gcc11.3.1-opt/geant4/11.1.2-brmbdz/lib64/l
ibG4run.so
#7  0x00007fc6c03ff823 in G4RunManager::InitializeGeometry() () from /cvmfs/sw-nightlies.hsf.org/key4hep/re
leases/2023-09-03/x86_64-almalinux9-gcc11.3.1-opt/geant4/11.1.2-brmbdz/lib64/libG4run.so
#8  0x00007fc6c03ff9b0 in G4RunManager::Initialize() () from /cvmfs/sw-nightlies.hsf.org/key4hep/releases/2
023-09-03/x86_64-almalinux9-gcc11.3.1-opt/geant4/11.1.2-brmbdz/lib64/libG4run.so
#9  0x00007fc6bb597d56 in SimG4Svc::initialize (this=0x36280d0) at /home/jsmiesko/Work/FCC/k4SimGeant4/SimG
4Components/src/SimG4Svc.cpp:72
```

**Comment by andresailer** [comment]
Hi @kjvbrt 
Did you also update k4FWCore with the change in IGeoSvc?

**Comment by kjvbrt** [comment]
hi @andresailer, I compiled branches `addAccessor` in both repos (k4FWCore first and after that k4SimGeant4)

**Comment by andresailer** [comment]
Where is "k4simgeant4env.sh" supposed to be?

https://github.com/HEP-FCC/k4SimGeant4/blob/90f21af70de2056ea9ac3da5e9691e539f935570/Detector/DetComponents/CMakeLists.txt#L30

Which is also used by that test in FCCDetectors.

**Comment by kjvbrt** [comment]
`k4simgeant4env.sh` is in build directory.

**Comment by andresailer** [comment]
Thanks @kjvbrt 
I was building everything together and that didn't create this file.
I found the offending `IGeoSvc.h` in this repository and removed it. Locally that test is passing now.

**Comment by kjvbrt** [comment]
great, merging.

**Pull request merged**